### PR TITLE
COLLECTED THINGS

### PR DESCRIPTION
**Pull request by Caele**
This is a draft for various things, but mainly the attribute expressions for coloring and labels. I will split these up into multiples before going ahead, but heres a sneak peak.

### CODE REVIEW

File: src/tree/render.js [hunk 1]
```diff
@@ -23,7 +23,12 @@ const filterTree = (id, tree) => {
       currentNode.data.id === node.data.id ||
       (currentNode.parent && node.data.id === currentNode.parent.data.id) ||
       (currentNode.parent && node.parent && node.parent.data.id === currentNode.parent.data.id) ||
-      (node.parent && node.parent.data.id === currentNode.data.id)
+      (node.parent && node.parent.data.id === currentNode.data.id) ||
+      (currentNode.parent &&
```
Comment (haxxmaxx) [inline src/tree/render.js]: Show the current node's sibling's children? Nieces I guess 😅

File: src/utils/tree-transform.js [hunk 2]
```diff
@@ -72,6 +74,35 @@ const getDataMatrix = async (layout, model) => {
...
+}
+
+function getAttribute(attrs, index) {
+  if (index !== false) {
+    return attrs.qValues[index].qText;
+  }
+  return false;
+}
+
+function transformColor(color) {
```
Comment (haxxmaxx) [inline src/utils/tree-transform.js]: We have a bunch of stuff in the button repo handling color, if it's needed :)

File: src/tree/render.js (hunk 1 above)
Comment (Caele) [reply src/tree/render.js]: yea, this part is not working properly anyway. Basically I wanted to show all subtrees without children as vertical stacks, but it breaks down for un-even trees
image

File: src/utils/tree-transform.js (hunk 2 above)
Comment (Caele) [reply src/utils/tree-transform.js]: yea, basically I just solved the ARGB case that engine spits out, but there are likely more things. And if we already solved them, we could pick them from there. 
This is probably something that should end up in nebula later.

Comment (haxxmaxx) [reply src/utils/tree-transform.js]: ah true that would be really nice

**Pull request closed without merging**